src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -912,19 +914,24 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
         {
             path_to_size_file[index] = (char)(48 + i);
 
-            // Only accept reading cache sizes from size files if they are 
-            // non-bogus values i.e., non-zero and != SIZE_MAX.
-            if (ReadMemoryValueFromFile(path_to_size_file, &size) && ((size != SIZE_MAX) || (size != 0)))
+            uint64_t cache_size_from_sys_file = 0;
+
+            if (ReadMemoryValueFromFile(path_to_size_file, &cache_size_from_sys_file))
             {
+                // uint64_t to long conversion as ReadMemoryValueFromFile takes a uint64_t* as an argument for the val argument.
+                size = (long)cache_size_from_sys_file;
                 path_to_level_file[index] = (char)(48 + i);
 
                 if (ReadMemoryValueFromFile(path_to_level_file, &level))
                 {
                     UPDATE_CACHE_SIZE_AND_LEVEL(size, level)
                 }
-                else
+
+                // We guard against the case where the size = -1 or 0. 
+                // This isn't currently an issue but to be safe, we add this check.
+                else if (size > 0)
```

Comment:
I don't think we need this guard here. 0 size would be ok and the first `ReadMemoryValueFromFile` should fail when it cannot extract a size.

Reply:
CC: @Maoni0 - we were discussing that this was a redundant check, but it'll be still good to add to prevent issues where the underlying mechanism to get cache size information changes. Such as in this case where we didn't know that the sysconf mechanism was different. 

I don't feel too strongly about this - please let me know what you think, and I can change this code accordingly.

Reply:
right, it would guard against both 0 and -1. also the comment really should just be deleted.

looking at this function, it seems like we have some inconsistency between this for loop and the check below that says `if (cacheLevel != 3)`, presumably the for loop is saying we can get a level 4 cache that is valid yet it will be overwritten by the `if (cacheLevel != 3)` code below since it's not 3. but that's a different issue.